tests/unit/Provider/JenssegersAgentTest.php

```diff
@@ -0,0 +1,248 @@
+<?php
+namespace UserAgentParserTest\Unit\Provider;
+
+use UserAgentParser\Provider\JenssegersAgent;
+
+/**
+ * @covers UserAgentParser\Provider\JenssegersAgent
+ */
+class JenssegersAgentTest extends AbstractProviderTestCase
+{
+    /**
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getParser()
+    {
+        $parser = $this->getMock('Jenssegers\Agent\Agent', [], [], '', false);
+
+        return $parser;
+    }
+
+    public function testPackageNotLoadedException()
+    {
+        $file     = 'vendor/jenssegers/agent/composer.json';
+        $tempFile = 'vendor/jenssegers/agent/composer.json.tmp';
+
+        rename($file, $tempFile);
+
+        try {
+            $provider = new JenssegersAgent();
+        } catch (\Exception $ex) {
+            // we need to catch the exception, since we need to rename the file again!
+        }
+
+        $this->assertInstanceOf('UserAgentParser\Exception\PackageNotLoadedException', $ex);
```

Comment:
if i move `rename()` to catch, the test won't fail. 
Can you give me an example how you would write it?